src/lit-html.ts

```diff
@@ -194,24 +192,23 @@ export class Template {
         // attributes are not guaranteed to be returned in document order. In
         // particular, Edge/IE can return them out of order, so we cannot assume
         // a correspondance between part index and attribute index.
-
-        // Do a first pass to count attributes that correspond to parts.
-        const attributesWithParts: string[][] = Array.prototype.filter.call(
-            attributes,
-            (attribute: Attr) =>
-                attribute.value.split(attrOrTextRegex).length > 1);
-        // Loop that many times, but don't use loop index for anything.
-        for (let i = 0; i < attributesWithParts.length; i++) {
+        let count = 0;
+        for (let i = 0; i < attributes.length; i++) {
+          if (attributes[i].value.indexOf(marker) >= 0) {
+            count++;
+          }
+        }
+        while (count--) {
           // Get the template literal section leading up to the first
           // expression in this attribute attribute
           const stringForPart = strings[partIndex];
           // Find the attribute name
           const attributeNameInPart =
-              stringForPart.match(lastAttributeNameRegex)![1];
+              lastAttributeNameRegex.exec(stringForPart)![1];
```

Comment:
Since `Symbol.match` came around, using `String#match` has gotten slower. Instead, just do the thing `#match` does directly, and save a byte. 😄

Reply:
Nice :)